pkg/auth/handler/webapp/authflowv2/reset_password_success.go

```diff
@@ -44,7 +45,61 @@ func (h *AuthflowV2ResetPasswordSuccessHandler) GetData(w http.ResponseWriter, r
 	return data, nil
 }
 
+func (h *AuthflowV2ResetPasswordSuccessHandler) GetNonAuthflowData(w http.ResponseWriter, r *http.Request) (map[string]interface{}, error) {
+	data := make(map[string]interface{})
+
+	baseViewModel := h.BaseViewModel.ViewModel(r, w)
+	resetPasswordSuccessViewModel := &AuthflowV2ResetPasswordSuccessViewModel{
+		CanBackToSignIn: r.URL.Query().Get("x_can_back_to_login") == "true",
+	}
+	viewmodels.Embed(data, baseViewModel)
+	viewmodels.Embed(data, resetPasswordSuccessViewModel)
+
+	return data, nil
+}
+
 func (h *AuthflowV2ResetPasswordSuccessHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	if isURLFromAdminAPI(r) {
+		h.serveHTTPNonAuthflow(w, r)
+	} else {
+		h.serveHTTPAuthflow(w, r)
+	}
+}
+
+func (h *AuthflowV2ResetPasswordSuccessHandler) serveHTTPNonAuthflow(w http.ResponseWriter, r *http.Request) {
+	makeHTTPHandler := func(handler func(w http.ResponseWriter, r *http.Request) error) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			var err error
+			err = handler(w, r)
+			if err != nil {
+				if apierrors.IsAPIError(err) {
+					// Still uses AuthflowController to render error because of same logic
+					h.Controller.ErrorRenderer.RenderError(w, r, err)
+				} else {
+					panic(err)
+				}
+			}
+		})
+	}
+
+	getHandler := makeHTTPHandler(func(w http.ResponseWriter, r *http.Request) error {
+		data, err := h.GetNonAuthflowData(w, r)
+		if err != nil {
+			return err
+		}
+		h.Renderer.RenderHTML(w, r, TemplateWebAuthflowResetPasswordSuccessHTML, data)
+		return nil
+	})
+	switch r.Method {
+	case "GET":
+		getHandler.ServeHTTP(w, r)
+	default:
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+}
+
+func (h *AuthflowV2ResetPasswordSuccessHandler) serveHTTPAuthflow(w http.ResponseWriter, r *http.Request) {
```

Comment:
Originally, it is already using `HandleWithoutFlow` which does not assume it is in an authflow, so I think we don't need to modify this handler.